website/docs/topics/feature-flags.md

```diff
@@ -0,0 +1,32 @@
+## 10 Principles for building and scaling feature flag systems
+
+Feature flags, sometimes called feature toggles or feature switches, are a software development technique that allows engineering teams to decouple the release of new functionality from software deployments. With feature flags, developers can turn specific features or code segments on or off at runtime, without the need for a code deployment or rollback.d Organizations who adopt feature flags see improvements in all key operational metrics for DevOps: Lead time to changes, mean-time-to-recovery, deployment frequency, and change failure rate.
+
+There are 10 principles for building a large-scale feature flag system. These principles have their roots in distributed systems architecture and pay particular attention to security, privacy, and scale that is required by most enterprise systems. If you follow these principles, your feature flag system is less likely to break under load and will be easier to evolve and maintain.
+
+These principles are:
+
+1. Enable run-time control. Control flags dynamically, not using config files.
+2. Design for failure. Favor availability over consistency.
+3. Use unique names across all applications. Enforce naming conventions.
+4. Evaluate locally. Evaluate flags as close to the user as possible.	
+5. Make feature flags short-lived. Do not confuse flags with application configuration.
+6. Scale horizontally. Decouple writing and reading flags.
+7. Limit payloads. Feature flag payload should be as small as possible.
+8. Choose open by default. Democratize feature flag access.	
+9. Do no harm. Prioritize consistent user experience.
+10. Enable traceability. Make it easy to understand flag evaluation.
+
+#### Background
```

Comment:
Headings should always follow in logical order. 1 -> 2 -> 3 -> 4. Skipping from 1 or 2 to 4 causes issue with the hierarchy.

```suggestion
## Background
```